<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>733</width>
    <height>381</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Connect Four</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="acceptDrops">
    <bool>false</bool>
   </property>
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>43</x>
      <y>9</y>
      <width>431</width>
      <height>261</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="currentIndex">
     <number>3</number>
    </property>
    <widget class="QWidget" name="pageGameView">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <widget class="GameView" name="gameView">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>101</width>
        <height>101</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="renderHints">
       <set>QPainter::SmoothPixmapTransform|QPainter::TextAntialiasing</set>
      </property>
      <property name="resizeAnchor">
       <enum>QGraphicsView::AnchorViewCenter</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="pageHighscore">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <widget class="HighscoreTableView" name="highscoreTableView">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>271</width>
        <height>161</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="pageLoadGame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <widget class="LoadGameTableView" name="loadGameTableView">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>256</width>
        <height>192</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="pageSqlAdmin"/>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>733</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuConnect_Four">
    <property name="title">
     <string>Connect Four</string>
    </property>
    <addaction name="actionNew_Game"/>
    <addaction name="actionLoad_game"/>
    <addaction name="actionSave_game"/>
    <addaction name="separator"/>
    <addaction name="actionPrint_higscore"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <addaction name="actionToggle_Fullscreen"/>
    <addaction name="separator"/>
    <addaction name="actionPrimitives_Design"/>
    <addaction name="actionSimple_Design"/>
    <addaction name="actionSvg_Design"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionHighscore"/>
    <addaction name="actionGame_board"/>
   </widget>
   <addaction name="menuConnect_Four"/>
   <addaction name="menuView"/>
   <addaction name="menuOptions"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionToggle_Fullscreen">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Toggle Fullscreen</string>
   </property>
   <property name="shortcut">
    <string>Alt+Return</string>
   </property>
  </action>
  <action name="actionNew_Game">
   <property name="text">
    <string>New Game</string>
   </property>
  </action>
  <action name="actionHighscore">
   <property name="text">
    <string>Highscore</string>
   </property>
  </action>
  <action name="actionLoad_game">
   <property name="text">
    <string>Load game</string>
   </property>
  </action>
  <action name="actionSave_game">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save game</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionGame_board">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Game board</string>
   </property>
  </action>
  <action name="actionPrimitives_Design">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Abstract Design</string>
   </property>
  </action>
  <action name="actionSimple_Design">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simple Design</string>
   </property>
  </action>
  <action name="actionSvg_Design">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Fancy Design</string>
   </property>
  </action>
  <action name="actionPrint_higscore">
   <property name="text">
    <string>Print higscore</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GameView</class>
   <extends>QGraphicsView</extends>
   <header>gameview.h</header>
   <slots>
    <slot>selectSimpleDesign(bool)</slot>
    <slot>selectPrimitivesDesign(bool)</slot>
    <slot>selectSvgDesign(bool)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>HighscoreTableView</class>
   <extends>QTableView</extends>
   <header>highscoretableview.h</header>
  </customwidget>
  <customwidget>
   <class>LoadGameTableView</class>
   <extends>QTableView</extends>
   <header>LoadGameTableView.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionNew_Game</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openNewGameDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionHighscore</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openHighscore()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_game</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openLoadGame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_game</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveGame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPrimitives_Design</sender>
   <signal>toggled(bool)</signal>
   <receiver>gameView</receiver>
   <slot>selectPrimitivesDesign(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>93</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimple_Design</sender>
   <signal>toggled(bool)</signal>
   <receiver>gameView</receiver>
   <slot>selectSimpleDesign(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>93</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSvg_Design</sender>
   <signal>toggled(bool)</signal>
   <receiver>gameView</receiver>
   <slot>selectSvgDesign(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>93</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggle_Fullscreen</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleFullscreen(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGame_board</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openGameBoard()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPrint_higscore</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>printHighscore()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>toggleFullscreen(bool)</slot>
  <slot>openNewGameDialog()</slot>
  <slot>openHighscore()</slot>
  <slot>openLoadGame()</slot>
  <slot>saveGame()</slot>
  <slot>openGameBoard()</slot>
  <slot>printHighscore()</slot>
 </slots>
</ui>
